apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnection.cs

```diff
@@ -489,7 +514,7 @@ private Task<Session> AllocateSession(TransactionOptions options, CancellationTo
                             _sharedSessionAllocator = SpannerClient.CreateSessionFromPoolAsync(
                                 _connectionStringBuilder.Project, _connectionStringBuilder.SpannerInstance,
                                 _connectionStringBuilder.SpannerDatabase, options, CancellationToken.None);
-                            _sharedSessionAllocator.ContinueWith(t =>
+                           result = _sharedSessionAllocator.ContinueWith(t =>
```

Comment:
ContinueWith can be dangerous - see https://blog.stephencleary.com/2015/01/a-tour-of-task-part-7-continuations.html

Can we use await instead here?